server/data/data-library/src/main/java/org/apache/james/rrt/lib/CanSendFromImpl.java

```diff
@@ -60,4 +77,24 @@ private boolean emailIsAnAliasOfTheConnectedUser(Username connectedUser, Usernam
             .anyMatch(alias -> alias.equals(connectedUser));
     }
 
+    private Stream<String> relatedAliases(Username user) throws RecipientRewriteTableException {
+        return Stream.concat(
+            Stream.of(user.getLocalPart()),
+            recipientRewriteTable.listSources(Mapping.alias(user.asString()))
+                .map(MappingSource::getFixedUser)
+        );
+    }
+
+    private Stream<String> relatedDomains(Username user) {
+        return user.getDomainPart()
+            .map(Throwing.function(this::fetchDomains).sneakyThrow())
+            .orElseGet(Stream::empty);
+    }
+
+    private Stream<String> fetchDomains(Domain domain) throws RecipientRewriteTableException {
+        return Stream.concat(
+          Stream.of(domain.asString()),
+          recipientRewriteTable.listSources(Mapping.domain(domain)).map(MappingSource::getFixedDomain)
```

Comment:
The indent looks weird

Reply:
@rouazana can you explain it please?

Reply:
oh maybe the issue is that it's a 2 spaces indent instead of a 4 spaces indent?